Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -147,20 +153,15 @@ async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishin
     }
     catch(error) {
         tl.error(tl.loc('Failedtodeploywebsite'));
-        isDeploymentSuccess = false;
-        deploymentError = error;
+        throw Error(error);
     }
 
     try {
-        tl._writeLine(await azureRESTUtility.updateDeploymentStatus(publishingProfile, isDeploymentSuccess));
+        tl._writeLine(await azureRESTUtility.updateDeploymentStatus(publishingProfile, true));
```

Comment:
This utility is being used for machine group task as well, don't move the update deployment history failure scenarios from here.